typing/typecore.ml

```diff
@@ -5839,113 +5952,178 @@ and type_let ?check ?check_strict
   let spatl =  List.map vb_pat_constraint spat_sexp_list in
   let attrs_list = List.map fst spatl in
   let is_recursive = (rec_flag = Recursive) in
-
-  let (pat_list, exp_list, new_env, mvs, _pvs) =
+  let (pat_list, new_env, exp_list, mvs, pvs) =
```

Comment:
Same problem here.
The changes are too extensive for a fast review.
More information about what was done and why would help.